src/components/ElementBar/ColorPickerButton/calculateColorChangeUpdates.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2024 Nordeck IT + Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Elements } from '../../../state/types';
+
+/**
+ * Calculates the updates to update element colours.
+ *
+ * @param elements - Elements to update the colours of
+ * @returns List of updates to apply the colour
+ */
+export const calculateColorChangeUpdates = (
+  elements: Elements,
+  color: string,
+) => {
```

Comment:
1. This should be simply a function
2. Please use explicit return types as already mentioned in previous review
3. Please use `ElementUpdate` that was requested and added by you already

Reply:
Types added